[9.0] Mis builder Report : Belgian VAT Declaration

vdewulf opened the pull request:
This PR adds a new template in the MIS Builder Templates for Belgium : the VAT Declaration. It is based on the tags on the taxes. The translation has been made in French only.
Improvement to come : add the series of logical and arithmetical checks that are usually provided by accounting tools.

It has been validated with an accounting expert for the more usual grids (01, 03, 49, 81, 82, 85, 54, 63, 59, 64, 71/72).

@sbidoul here is my contribution for the VAT report

Commit:
[ADD] l10n_be_mis_reports: new vat data

Commit:
[IMP] replaced bal[%] by bal[] in KPI expressions

Commit:
[IMP] Added a condition to display only the positive amount
in grid 71 and grid 72

Commit:
[IMP] updated the readme, the french translation and removed one line
in VAT declaration (one arithmetical check).
Waiting for the complete set of checks to add them at once.

sbidoul commented on l10n_be_mis_reports/data/mis_report_vat.xml:
```diff
@@ -0,0 +1,369 @@
+      <field name="report_id" ref="mis_report_vat"/>
+      <field name="name">grid_48</field>
+      <field name="description">Grid 48</field>
+      <field name="expression">bal[][('tax_ids.tag_ids.name', 'like', '% 48'),('debit','>',0)] </field>
```
For cases like this where we want only debits, it's better to use `deb[][('tax_ids.tag_ids.name', 'like', '% 48')]`. Performance will be better.

Commit:
[IMP] replaced the bal expressions with "deb" and "crd"

vdewulf commented:
@sbidoul Thanks a lot for the review. I replaced all the "bal" expressions with "deb" and "crd" according to your advice.

sbidoul committed:
[IMP] add menu entry for VAT declaration

sbidoul commented:
@vdewulf I created https://github.com/vdewulf/l10n-belgium/pull/1 to add your report as a menu.

Commit:
Merge pull request #1 from acsone/9.0-mis-report-vat-sbi

[IMP] add menu entry for VAT declaration

sbidoul commented on l10n_be_mis_reports/__openerp__.py:
```diff
@@ -39,6 +39,7 @@
+        'data/mis_report_vat.xml',
     ],
     'installable': True,
 }
```
Now that we depends on the menu (l10n_be.account_reports_be_statements_menu) we need to add the dependency on l10n_be here.

Commit:
[ADD] Some logical and arithmetical controls added

Commit:
[ADD] add dependency on l10n_be

vdewulf commented:
@sbidoul I added the dependency as suggested.
Thanks!

sbidoul commented on l10n_be_mis_reports/data/mis_report_vat.xml:
```diff
@@ -0,0 +1,472 @@
+      <field name="description">Control Grid 87: [85] + [87] x 0.21 >= [56] + [57]</field>
+      <field name="expression">min((grid_85 + grid_87)*0.21-grid_57-grid_56,AccountingNone)</field>
+      <field name="type">num</field>
+      <field name="compare_method">pct</field>
```
Very very nice you add these checks.

Would you like to try this kind of expression (using type=str) ?

`'✔' if (grid_85 + grid_87)*0.21 >= (grid_57 + grid_56) else '✘'`

Or do you think having the actual difference in the cell is better?

sbidoul commented on l10n_be_mis_reports/data/mis_report_vat.xml:
```diff
@@ -0,0 +1,472 @@
+      <field name="report_id" ref="mis_report_vat"/>
+      <field name="name">grid_64</field>
+      <field name="description">Grid 64</field>
+      <field name="expression">deb[][('tax_line_id.tag_ids.name', 'like', '% 64')]</field>
```
Now that mis_builder version 3.1 is merged, it is possible to write

`deb[][('tax_line_id.tag_ids.id', '=', ref('l10n_be.tax_tag_64').id)]`

This would be more robust in case someone decides to translate or otherwise change the tag labels. 

Not blocking for me though.

Commit:
[IMP] Updated the expressions to use the ref on the tags
thanks to upgrade of MIS Builder to v3.1
Updated the controls to Strings expressions with 'V' and 'X'

Commit:
[IMP] Correction of controls (missing "()")
Update of french translation with the corrections

vdewulf commented:
@sbidoul I validated the controls with an accounting expert.
In principles, my work is done on this topic.
Could you consider merging this?
Thanks a lot!

sbidoul commented:
@vdewulf thanks a lot for this fantastic work.

In principle we should need another review but since we are a small population here, I'll merge.

Note I'll soon work on v10 and the intracom declaration.
